crates/allocator/src/bump.rs

```diff
@@ -69,29 +69,33 @@ impl InnerAlloc {
             start
         } else {
             let prev_page = core::arch::wasm32::memory_grow(0, 1);
-            if prev_page == usize::MAX() {
+            if prev_page == usize::MAX {
                 return core::ptr::null_mut()
             }
 
-            let start = prev_page
-                .checked_mul(PAGE_SIZE)
-                .unwrap_or_else(|| return core::ptr::null_mut());
+            let start = match prev_page.checked_mul(PAGE_SIZE) {
+                Some(s) => s,
+                None => return core::ptr::null_mut(),
+            };
```

Comment:
To help with code like this you could define another implementation function for this function with a return type of `Option<*mut u8>` instead of `*mut u8` and convert the `None` case to a `core::ptr::null_mut()`. Then you could use `?` syntax in the implementation function. Then the `alloc` function would simply act as a thin wrapper around this. This might bring other downsides of which I am unaware of, yet.